swap/src/cli/command.rs

```diff
@@ -765,6 +771,20 @@ mod tests {
         assert_eq!(args, Arguments::resume_testnet_defaults().with_debug());
     }
 
+    #[test]
+    fn given_help_does_not_panic() {
+        let raw_ars = vec![BINARY_NAME, "--help"];
```

Comment:
How does this work? there is no `help` field in `RawArguments`? Doesn't `--help` have to "applied" to another argument?

Reply:
> > If we do that we basically re-implement parts of clap - is that really what we want to do?
> 
> I think that is the price you pay if you want more explicit control over the parsing. Literally killing the process is much more surprising behaviour that may be very hard to find / spot.
> 
> Also "re-implement" makes it sound like you will have to write a 100 lines of code. A downcast of the error and a pattern match should sort you.

unfortunately it is not that simple - If I do this:

```
match parse_args_and_apply_defaults(env::args_os()) {
        Ok(args) => args,
        Err(e) => {
            if let Some(clap_err) = e.downcast_ref::<clap::Error>() {
                match clap_err.kind {
                    ErrorKind::HelpDisplayed | ErrorKind::VersionDisplayed => {
                        std::process::exit(0);
                    }
                    _ => {
                        bail!(e);
                    }
                }
            }
            bail!(e);
        }
    };
```

then `--help` does not print anything. So I would have to somehow manually trigger the help / version display as well.

Reply:
I reckon I can handle this inside the parsing function where I have access to clap - and could trigger `write_help`. But then I don't see much advantages over using `get_matches_from` because we would have to handle it in the function anyway.